### DISCUSSION TITLE
Changes detection

### DISCUSSION THREAD

**Discussion by vadimkantorov** [facebook/lexical#6312] [General]
Is there a way of obtaining an estimate of "any changes detection" from the editor instance? (e.g. at least if no edits were done whatsoever)

Even a conservative dirty detection would be a useful API (e.g. for determining if auto-saves (or suggesting a save to the user) to a remote storage should be performed)

Or should manual checking of editor state equivalence to the initial state work?

**Comment by 2wheeh** [comment]
There is onChangePlugin. I would auto save by passing debounced callback to onChangePlugin.

**Reply by vadimkantorov** [reply]
And in general, is there an efficient way to track any difference from the fixed initial editor state? (so that typing a few letters and then deleting them would indicate no changes)

Comparing a big editor state tree to the initial tree at every keystroke/change probably would lead to quite laggy typing

**Reply by vadimkantorov** [reply]
@2wheeh What do you mean by debounced callback?

I've tried using `OnChangePlugin`, but it gets triggered even when I move caret around the document. It appears that `ignoreSelectionChange` from [docs](https://lexical.dev/docs/react/plugins#lexicalonchangeplugin) does not apply to ignoring moving cursor/caret (it appears that this condition is not working for some reason to skip cursor place changes: `dirtyElements.size === 0 && dirtyLeaves.size === 0)`) - rendering it not very useful for change tracking

**Reply by vadimkantorov** [reply]
Oops, my bad, `ignoreSelectionChange={true}` makes OnChangePlugin ignore cursor changes
